cloudify_rest_client/secret_provider.py

```diff
@@ -0,0 +1,68 @@
+from cloudify_rest_client.responses import ListResponse
+from cloudify_rest_client.constants import VisibilityState
+
+
+class SecretProvider(dict):
+    def __init__(self, secret):
+        super(SecretProvider, self).__init__()
+        self.update(secret)
+
+    @property
+    def name(self):
+        return self.get('name')
+
+    @property
+    def type(self):
+        return self.get('type')
+
+    @property
+    def connection_parameters(self):
+        return self.get('connection_parameters')
+
+    @property
+    def created_at(self):
+        return self.get('created_at')
+
+    @property
+    def updated_at(self):
+        return self.get('updated_at')
+
+    @property
+    def visibility(self):
+        return self.get('visibility')
+
+    @property
+    def tenant_name(self):
+        return self.get('tenant_name')
+
+
+class SecretProviderClient(object):
+    def __init__(self, api):
+        self.api = api
+
+    def create(
+            self,
+            secret_provider_name,
+            secret_provider_type,
+            connection_parameters,
+            tenant_name,
```

Comment:
```suggestion
            tenant_name=None,
```
this must not be required, but needs to be defaulted (from the client's header)

Reply:
actually nope, just remove `tenant_name` from here entirely (and don't pass it through from the CLI).

The restclient object, when it's created, already has the selected tenant name in its headers